src/omv/imlib/jpeg_dec.c

```diff
@@ -2420,22 +2388,37 @@ static void JPEGPutMCUGray(JPEGIMAGE *pJPEG, int x, int y)
         usDest += iPitch; // next line
     }
 } /* JPEGPutMCUGray() */
+#endif // NOT_USED
 
 static void JPEGPixelLE(uint16_t *pDest, int iY, int iCb, int iCr)
 {
     int iCBB, iCBG, iCRG, iCRR;
+    int tmp;
     unsigned short usPixel;
     
     iCBB = 7258  * (iCb-0x80);
     iCBG = -1409 * (iCb-0x80);
     iCRG = -2925 * (iCr-0x80);
     iCRR = 5742  * (iCr-0x80);
-    usPixel = usRangeTableB[((iCBB + iY) >> 12) & 0x3ff]; // blue pixel
-    usPixel |= usRangeTableG[((iCBG + iCRG + iY) >> 12) & 0x3ff]; // green pixel
-    usPixel |= usRangeTableR[((iCRR + iY) >> 12) & 0x3ff]; // red pixel
+    tmp = (iCBB + iY) >> 15;
+    if (tmp < 0) tmp = 0;
+    else if (tmp > 31) tmp = 31; // blue
+    usPixel = tmp;
+    tmp = ((iCBG + iCRG + iY) >> 14);
+    if (tmp < 0) tmp = 0;
+    else if (tmp > 63) tmp = 63;
```

Comment:
This is literally the __USAT instruction with the shift part.

Reply:
Yes, that particular operation is USAT16. The earlier table version was useful because it did the final shift into position for RGB565. I'll replace the range checking with USAT in my next pass.